src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_TryCatch.vb

```diff
@@ -104,5 +110,742 @@ IBlockStatement (3 statements) (OperationKind.BlockStatement) (Syntax: 'Private
+Imports System
+Class C
+    Private Sub M(e As Exception)
+        Try
+        Catch e'BIND:"e"
+        End Try
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IParameterReferenceExpression: e (OperationKind.ParameterReferenceExpression, Type: System.Exception) (Syntax: 'e')
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of IdentifierNameSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/22299")>
+        Public Sub TryCatch_GetOperationForCatchExceptionDeclaration()
+            Dim source = <![CDATA[
+Imports System
+Class C
+    Private Sub M()
+        Try
+        Catch e As Exception'BIND:"e"
+        End Try
+    End Sub
+End Class]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration) (Syntax: 'e')
+  Variables: Local_1: e As System.Exception
+  Initializer: null
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of IdentifierNameSyntax)(source, expectedOperationTree, expectedDiagnostics)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub TryCatch_GetOperationForCatchStatement()
+            Dim source = <![CDATA[
+Imports System
+Class C
+    Private Shared Sub M()
+        Try
+        Catch e As IO.IOException When e.Message IsNot Nothing'BIND:"Catch e As IO.IOException When e.Message IsNot Nothing"
+        End Try
+    End Sub
+End Class]]>.Value
+
+            ' GetOperation returns Nothing for CatchStatementSyntax
+            Assert.Null(GetOperationTreeForTest(Of CatchStatementSyntax)(source).operation)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub TryCatch_GetOperationForTryStatement()
+            Dim source = <![CDATA[
+Imports System
+Class C
+    Private Shared Sub M()
+        Try'BIND:"Try"
+        Catch e As IO.IOException When e.Message IsNot Nothing
+        End Try
+    End Sub
+End Class]]>.Value
+
+            ' GetOperation returns Nothing for TryStatementSyntax
+            Assert.Null(GetOperationTreeForTest(Of TryStatementSyntax)(source).operation)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact>
+        Public Sub TryCatch_GetOperationForFinallyStatement()
+            Dim source = <![CDATA[
+Imports System
+Class C
+    Private Shared Sub M(s As String)
+        Try
+        Finally'BIND:"Finally"
+            Console.WriteLine(s)
+        End Try
+    End Sub
+End Class]]>.Value
+
+            ' GetOperation returns Nothing for FinallyStatementSyntax
+            Assert.Null(GetOperationTreeForTest(Of FinallyStatementSyntax)(source).operation)
```

Comment:
>GetOperationTreeForTest [](start = 24, length = 23)

It would be good to test what nodes do we get for EndTryStatement, CatchFilterClause, CatchFilterClause.Filter and for an individual statement inside Try/Catch/Finally blocks.